src/options.cpp

```diff
@@ -180,6 +181,9 @@ options parse_options(int argc, char** argv) {
       } else {
         next_vim_file_bufnr = bufnr;
       }
+    } else if (parser.match_flag_option("--help"sv, "--h"sv)) {
+      o.help = true;
+      goto done_parsing_options;
```

Comment:
Seems kinda weird that we stop parsing options here. I'd continue parsing options after `--help`.